client/solo/commands/deptest.py

```diff
@@ -207,12 +207,17 @@ def _test(self, entropy_client):
             entropy_client.output(mytxt, header=darkred(" @@ "))
             time.sleep(10)
 
+        # avoid saving deps as installed by user
+        matched_for_installation = self._match_packages_for_installation(
+            entropy_client, False, sorted(found_deps))
+        if not matched_for_installation:
+            return 1
         exit_st, _show_cfgupd = self._install_action(
             entropy_client, True, True,
             self._pretend, self._ask,
             False, self._quiet, False,
             False, False, False, False, False,
-            False, 1, sorted(found_deps))
```

Comment:
Nope, _match_packages_for_installation will call _scan_packages itself.

I could have explained this better (and will update the commit message if this is agreed):

I made it explicit in documentation (I didn't change that logic) that in _install_action
````
 +        Packages passed in the packages argument (as opposed to
 +        package_matches) will be marked as installed by user. 
```
and that's why here I am passing the already scanned packages to `package_matches` because deptest isn't supposed to mark installed packages as by-user.

In general, my changes make use of the currently existing logic, with the improvement that the behaviour of _install_packages is now more explicit.